### PR TITLE
John Roemer code challenge week 3

### PR DESCRIPTION
**Pull request by jtrtj**
Only completed 2 user stores.

### CODE REVIEW

**Comment by jtrtj** [comment]
@iandouglas @dionew1 
I need some help - I can't figure out why my features test keeps telling me `@book` is nil but when I pry around it doesn't seem to be and my model tests pass for the same methods that my feature tests says return nil.

File: app/models/book.rb [hunk 1]
```diff
@@ -1,3 +1,24 @@
 class Book < ApplicationRecord
-  
+  has_many :reviews
+  has_many :users, through: :reviews
```
Comment (dionew1) [inline app/models/book.rb]: Nice job setting up these relationships!

File: app/models/book.rb [hunk 2]
```diff
@@ -1,3 +1,24 @@
...
+  end
+
+  def highest_rating
+    reviews.maximum(:rating)
+  end
+
+  def lowest_rating
+    reviews.minimum(:rating)
+  end
 end
```
Comment (dionew1) [inline app/models/book.rb]: Great job using ActiveRecord here in the instance methods. 😄 
It would appear that you are not using the methods highet_rating or lowest_rating, so you can remove them from your model.

File: app/views/books/index.html.erb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<body>
+  <h1>Books</h1>
+  <ul>
+    <% @books.each do |book| %>
+      <li><%= link_to "#{book.title}", book_path(book) %></li>
```
Comment (dionew1) [inline app/views/books/index.html.erb]: You don't have to interpolate here you can just use `book.title`.

File: spec/features/book_features_spec.rb [hunk 4]
```diff
@@ -0,0 +1,69 @@
...
+    it 'will see a list of all books and their titles will be links to show pages' do
+      bk_1 = Book.create(title: 'sdfg')
+      bk_2 = Book.create(title: 'hgfd')
+
+      visit books_path
+
+      expect(page).to have_content(bk_1.title)
+      expect(page).to have_content(bk_2.title)
+
+      click_on "#{bk_1.title}"
```
Comment (dionew1) [inline spec/features/book_features_spec.rb]: We can also just use `bk_1.title` here too.

File: spec/features/book_features_spec.rb [hunk 5]
```diff
@@ -0,0 +1,69 @@
+require 'rails_helper'
+
+describe 'visitor' do
+  context 'visiting books index' do
+    it 'will see a list of all books and their titles will be links to show pages' do
+      bk_1 = Book.create(title: 'sdfg')
+      bk_2 = Book.create(title: 'hgfd')
+
```
Comment (dionew1) [inline spec/features/book_features_spec.rb]: Since we are visiting the book show page within this test it is running through the html and hitting the @highest_review and getting nil which caused this test to fail. You would need to add a user and review in the setup of this test in order to get it passing again.
